airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/parquet_parser.py

```diff
@@ -126,7 +126,7 @@ def _scalar_to_python_value(parquet_value: Scalar, parquet_format: ParquetFormat
 
         if pa.types.is_decimal(parquet_value.type):
             if parquet_format.decimal_as_float:
-                return parquet_value.as_py()
+                return float(parquet_value.as_py())
```

Comment:
Change: Pydantic v1 had the default behavior to cast decimals to floats. In v2, their default behavior is to cast decimals to strings (similar to what we're made our default, likely for the same loss of precision reason). This means that in the default pydantic behavior, `decimal_as_float=True` would have the same result as `decimal_as_float=False`

To continue to allow `decimal_as_float` (we may want to remove this later, but don't need to right now), we cast it to float ourselves before passing it to the pydantic v2 model. This also more accurately represents whats happening with the parameter name `decimal_as_float` :)

This is for the `parquet_file_with_decimal_as_float` scenario.